### PR TITLE
Log some semi-helpful messages if the graph page fails a query request

### PR DESCRIPTION
**Pull request by kevmoo**
Fixes https://github.com/dart-lang/build/issues/1372

### CODE REVIEW

File: build_runner/lib/src/server/graph_viz_main.dart [hunk 1]
```diff
@@ -15,21 +16,42 @@ main() async {
   searchForm.onSubmit.listen((e) async {
     e.preventDefault();
     var query = searchBox.value;
-    _focus(query);
+    await _focus(query);
```
Comment (natebosch) [inline build_runner/lib/src/server/graph_viz_main.dart]: could also make the method not async instead

File: build_runner/lib/src/server/graph_viz_main.dart [hunk 2]
```diff
@@ -15,21 +16,42 @@ main() async {
...
+  String requestPath;
+  try {
+    var parts = query.split('|');
+    var package = parts[0];
+    var path = parts[1];
+    requestPath = path.startsWith('lib/')
+        ? 'packages/$package/${path.substring(4)}'
+        : '${path.split('/').skip(1).join('/')}';
+  } catch (e, stack) {
+    _error('The query you provided ($query) could not be parsed.', e, stack);
```
Comment (grouma) [inline build_runner/lib/src/server/graph_viz_main.dart]: What do the queries look like here? Should we wrap it in back tics?

File: build_runner/lib/src/server/graph_viz_main.dart (hunk 1 above)
Comment (kevmoo) [reply build_runner/lib/src/server/graph_viz_main.dart]: true!

File: build_runner/lib/src/server/graph_viz_main.dart (hunk 2 above)
Comment (kevmoo) [reply build_runner/lib/src/server/graph_viz_main.dart]: It's just an asset ID – I think. The parens should be fine.

**Pull request merged**